Source/BugsnagMetadata.m

```diff
@@ -86,15 +81,22 @@ - (void)addAttribute:(NSString *)attributeName
         if (value) {
             id cleanedValue = BSGSanitizeObject(value);
             if (cleanedValue) {
-                [self getTab:tabName][attributeName] = cleanedValue;
-            } else {
+                NSDictionary *section = [self getMetadata:tabName];
+                if (!section) {
+                    [[self dictionary] setObject:[NSMutableDictionary new] forKey:tabName];
+                    section = [self getMetadata:tabName];
+                }
+                [section setValue:cleanedValue forKey:attributeName];
+            }
+            else {
                 Class klass = [value class];
                 bsg_log_err(@"Failed to add metadata: Value of class %@ is not "
                             @"JSON serializable",
                             klass);
             }
-        } else {
-            [[self getTab:tabName] removeObjectForKey:attributeName];
+        }
+        else {
+            [[self getMetadata:tabName] removeObjectForKey:attributeName];
```

Comment:
nit: it'd be good to rename `tabName` to `section` for consistency across this file

Reply:
Agreed, I'd go further and say uses of "tab" should change across all files, if the vocabulary is changing across all notifiers.